### PR TITLE
BB-428 Backport lifecycle listing optimization

### PR DESCRIPTION
**Pull request by nicolas2bert**
1. Backport:
- [BB-352](https://scality.atlassian.net/browse/BB-352) **Already Reviewed**
- [BB-374](https://scality.atlassian.net/browse/BB-374) **Already Reviewed**
- [BB-380](https://scality.atlassian.net/browse/BB-380) **Already Reviewed**

2. Fix linter.
3. Skip transition tests because the Lifecycle Transition workflow is not supported yet in Ring S3C.
4. Artesca moved `forceLegacyListing` out of the `bucketProcessor` configuration section.

[BB-352]: https://scality.atlassian.net/browse/BB-352?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[BB-374]: https://scality.atlassian.net/browse/BB-374?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[BB-380]: https://scality.atlassian.net/browse/BB-380?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jonathan-gramain — APPROVED** [review]

**Comment by nicolas2bert** [comment]
ping

**Comment by nicolas2bert** [comment]
ping

File: tests/functional/lifecycle/LifecycleTaskV2-versioned.js
Comment (francoisferrand) [inline tests/functional/lifecycle/LifecycleTaskV2-versioned.js:1]: this commit should be reverted in 8.5, we don't want to disable these test in 8.x

Comment (nicolas2bert) [reply tests/functional/lifecycle/LifecycleTaskV2-versioned.js:1]: It should not be skipped in:
[w/8.6/improvement/BB-428/lifecycle-listings](https://github.com/scality/backbeat/tree/w/8.6/improvement/BB-428/lifecycle-listings) 
 [w/8.5/improvement/BB-428/lifecycle-listings](https://github.com/scality/backbeat/tree/w/8.5/improvement/BB-428/lifecycle-listings)

**Review by francoisferrand — APPROVED** [review]

**Comment by nicolas2bert** [comment]
@bert-e approve

**Pull request merged**